File: Day348/LargestSumSubarrayofSizeAtLeastK.go
package main

import "fmt"

func maxSumWithK(a []int64, n, k int64) int64 {
	var answer, temp1, temp2, i, t int64

	for i = 0; i < k; i++ {
		temp1 += a[i]
	}

	answer = temp1

	for i < n {
		temp1 += a[i]
		temp2 += a[i-k]
		t = min(t, temp2)
		answer = max(answer, temp1-t)
		i++
	}

	return answer
}

func max(a, b int64) int64 {
	if a > b {
		return a
	}
	return b
}

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func main() {
	var T int64
	fmt.Scan(&T)

	for t := int64(0); t < T; t++ {
		var n, k int64
		fmt.Scan(&n)

		a := make([]int64, n)
		for i := int64(0); i < n; i++ {
			fmt.Scan(&a[i])
		}

		fmt.Scan(&k)

		result := maxSumWithK(a, n, k)
		fmt.Println(result)
	}
}

File: Day348/LargestSumSubarrayofSizeAtLeastK.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
	public static void main(String[] args) throws IOException
	{
	        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t =
            Integer.parseInt(br.readLine().trim()); // Inputting the testcases
        while(t-->0)
        {
            long n = Long.parseLong(br.readLine().trim());
            long a[] = new long[(int)(n)];
            // long getAnswer[] = new long[(int)(n)];
            String inputLine[] = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) {
                a[i] = Long.parseLong(inputLine[i]);
            }
            long k = Long.parseLong(br.readLine().trim());
            
            Solution obj = new Solution();
            System.out.println(obj.maxSumWithK(a, n, k));
            
        }
	}
}


// } Driver Code Ends


//User function Template for Java


class Solution {
    
    public long maxSumWithK(long a[], long n, long k)
    {
        long answer = 0;
        long temp1 = 0, temp2 = 0, i =0, t = 0;
        while (i < k) {
            temp1 += a[(int)i++];
        }
        answer = temp1;
        while (i<n) {
            temp1 += a[(int)i];
            temp2 += a[(int)(i-k)];
            t = Math.min(t, temp2);
            answer = Math.max(answer, temp1-t);
            i++;
        }
        return answer;
    }
}